src/test/java/htsjdk/samtools/CRAMAllEncodingStrategiesTest.java

```diff
@@ -91,7 +132,13 @@ public void assertRoundTripFidelity(
             final SAMRecordIterator sourceIterator = sourceReader.iterator();
             final SAMRecordIterator targetIterator = copyReader.getIterator();
             while (sourceIterator.hasNext() && targetIterator.hasNext()) {
-                if (emitDetail) {
+                if (lenientEquality) {
+                    final SAMRecord sourceRec = sourceIterator.next();
+                    final SAMRecord targetRec = targetIterator.next();
+                    Assert.assertEquals(targetRec.getReadName(), sourceRec.getReadName());
```

Comment:
Can lenient mode also check the start/end positions of the read?

Reply:
Yes, added.